Updated requesting page (#887)
BitcoinDesign/Guide: guide/daily-spending-wallet/requesting.md

-->

# Requesting bitcoin
{:.no_toc} 

Requesting bitcoin involves creating and sharing a payment request with a sender. A payment request is a piece of information that tells a sender where and how much bitcoin to send someone. This page covers design considerations when requesting bitcoin with a [daily spending wallet]({{ "/guide/daily-spending-wallet/" | relative_url }}).

{% include /tip/open.html label="Payment request formats" icon="info" color="blue" %}

Payment request formats are different types of payment requests. Many exist and each has unique properties and use cases. This page does not cover individual formats. For that, check out our  [payment request formats]({{ "/guide/how-it-works/payment-request-formats/" | relative_url }}) page.

{% include /tip/close.html %}

---

<div class="glossary-toc" markdown="1">
* Table of contents
{:toc}
</div>

---

## Requesting entry point

> Creating payment requests should be simple and fast. In this wallet, users can initiate the send process from the home screen directly after entering the app. 

Author: Added

Having the users able to enter an amount from the home page will make initiating a request even faster as they can jump straight into a requesting flow.

{% include tip/recommendation.html %}

Avoid showing the users balance on the users home or requesting page. When requesting payments in person this could reveal the users balance to near by on-lookers negatively affecting the users privacy. More on this [here](https://d.elor.me/2021/11/hiding-wallet-balances/)

{% include tip/close.html %}

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance@2x.png"
   layout = "shadow"

>    caption = "Do not show the users balance on the same page they request from. This can reveal the users balance to nearby on-lookers when requesting in-person affecting their privacy."

Reviewer: ```suggestion
   caption = "Showing wallet balance on the same users request from affects their privacy by revealing it to nearby on-lookers"
```

Reviewer: Trying to shorten captions a bit

   alt-text = "Screen showing home screen showing the users balance."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Home.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Home@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Home@2x.png"
   layout = "shadow"

>    caption = "Make initiating a request simple and fast such as from the users home screen. Putting a number pad on the home screen can make the process even faster as users jump straight into a flow."

Reviewer: ```suggestion
   caption = "Putting a number pad on the home screen makes transacting faster as users jump straight into a flow."
```

   alt-text = "Screen showing the users home page with a number pad that can be used to initiate a request."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Entering an amount

> Once the user is on the page they can initiate a request from, the next thing to do is add an amount to be requested. Users should be able easily toggle between entering amounts in sats, bitcoin, or the users local currency. If the user has an amount entered and they switch units the input should be cleared to prevent users requesting an incorrect amount. More on entering amounts on our [Units & Symbols page]({{ "/guide/designing-products/units-and-symbols/" | relative_url }}).

Author: You could have sats entered and accidently switch to dollars and then enter a sats amount (huge difference). Would prevent this error. But I also do what you said though the local currency is shown under the main input so you shouldn't have to do that

It should also be possible for users to enter a zero-amount as there may be scenarios where the sender is deciding the amount to send. Zero-amount requests also limit both parties exposure to price volatility as the bitcoin price in fiat terms could change between requesting and payment being sent.

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro@2x.png"
   layout = "shadow"
   caption = "Let users request in their local currency."
   alt-text = "Screen showing home screen with a fiat euro input selected."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer: +1 on spacings, but why avoiding the shorthand? These are used widely in both crypto and fiat currencies products. Full name also takes a bit more real estate on the screen, which will be already tight if we add spacings

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Author: What would you put as the shorthand? I think having BTC is necessary and is pretty universally used at this point

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer 2: Why not use ₿?

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer 2: Agree with Stephen. Also, the left screen uses sats, but the right one uses BTC? Shouldn't both be sats (which might be appropriate for a daily spending wallet).

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer 2: And heads-up that in all the UI Kit screens and pages I worked on I used a space between the number and the currency symbol.

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer 2: So no response or action on this one?

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Author: Will return to this in follow up PR

   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Invoice.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   layout = "shadow"
   caption = "Let users share lightning invoices separately as not all senders may be able to read unified requests."
   alt-text = "Screen showing an invoice that is part of a unified request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/address.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/address@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/address@2x.png"
   layout = "shadow"
   caption = "On-chain addresses should also be able to be shared individually."
   alt-text = "Screen showing an on-chain address that is part of a unified request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

> ## Fees and staying online

Reviewer 2: Should we mention something about solutions being worked on to allow for offline receives? Like [PTLCs](https://protos.com/bitcoin-lightning-dev-fix-existential-problem-offline-crypto-payments/)?

> ## Fees and staying online

Author: Agree, though they notification has both of those elements in so wasn't sure how best to separate the content. I will add some content to the Lightning services page and link to that rather then put it here.

<div class="center" markdown="1">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/metadata.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/metadata@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/metadata@2x.png"
   width = 250
   height = 541
   caption = "Let users add metadata to their created payment requests."
   alt-text = "Screen showing a menu with various metadata options for a payment request."
   layout = "float-left-desktop -background -shadow"
%}

Ensure users can [backup]({{ "/guide/daily-spending-wallet/backup-and-recovery/landing-page/" | relative_url }}) this metadata to prevent them losing their transaction history.

## Sharing the request

Once the request is created, the next action is to share it. Below are ways in which bitcoin requests are commonly shared.

### QR code

[Quick response codes (QR)](https://en.wikipedia.org/wiki/QR_code) encode a payment request into a scannable graphic. QR codes should be large enough and have high contrast with your application's background so they can be easily scanned.

> Uppercasing payment request data will result in less complex, more easily scannable, QR codes. Another option to make QR codes easier to scan is having users screen brightness turn up when on the screen.

Author: Nah that should be fine to add, I imagine if a designer doesn't know they will act the dev who will understand

### Plaintext

Plaintext involves copying and sharing a payment request in text form. These should be shared over secure communication channels.

### Contactless

A contactless, or [Near-field communication](https://en.wikipedia.org/wiki/Near-field_communication) (NFC) allows requests to be shared wirelesses over short distances.

### Payment link

> Payment links use a BIP21 [URI]({{ "/guide/how-it-works/payment-request-formats/#uniform-resource-identifier-uris-schemes" | relative_url }}) `bitcoin:` which makes these readable by other bitcoin applications. These can be included as part of a button or hypertext.

Author: It should say hypertext link

   alt-text = "Screen showing a payment request being shared contactlessly with NFC."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentLink.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentLink@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentLink@2x.png"
   layout = "shadow"
   caption = "Payment links make it simple for senders to import a payment request into their app."
   alt-text = "Screen showing a payment link being opened by a wallet."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Payment addresses

Payment addresses, also known as [reusable payment requests]({{ "/guide/how-it-works/payment-request-formats/#invoice-protocols" | relative_url }}), are requests that can receive more than a single payment. These are not widely supported but offer unique use cases, features, and improved usability. 

Due to the unique use cases of payment addresses, they are created, managed and shared in a separate area within the app.

> An example of how a payment address could be used is as quickly shared request from a users profile, say to receive tips on their website. If you are using payment addresses you should let users know that other wallets may not support sending to them and that the user needs to be online to receive payments.

Author: Yeah agree, we need offline receives (called async payments now by the LSP community). Will return to this once that's rolled out

<div class="image-slide-gallery">

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress.png"

Reviewer 2: Should there be an extra item in this list for managing payment addresses?

Author: Wasn't sure if we should keep that for a dedicated page on this and just have this as a stub for them for now until its more widely adopted

Author: Agree, lets come back to this

   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress@2x.png"
   layout = "shadow"
   caption = "An example of a payment address use case is being able to quickly share it from your profile page."
   alt-text = "Screen showing the users settings with a QR icon in the top left that opens a payment address."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR@2x.png"
   layout = "shadow"
   caption = "If supporting payment addresses, warn users that other apps may not be able to pay it and that they need to have their app open to receive payments."
   alt-text = "Screen showing QR code of a payment address."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

---

> ## Advanced requesting options

Reviewer 2: ```suggestion
## Advanced options
```

When denominating invoices in fiat, custom expiries should be used to prevent exposure to price volatility.

If requesting a specific amount denominated in fiat, use a shorter expiry and have the invoice refresh each time it expiries. For a wallet designed for in-person payments, a 30 - 60 second expiry works well.

If there is no amount defined on the invoice, the invoice is denominated in bitcoin, or the invoice needs to be shared in a message, then use a longer expiry, like 24 hours.

While it is good to allow users to define their own custom expiries, remember that only advanced users will do this. Choose a sensible default that makes sense for your user-base and how the wallet is intended to be used.

For some background on situations the user might encounter with different invoice expiries, see this [blog post](https://d.elor.me/2022/01/lightning-invoice-expiration-ux-considerations/) from designer Stephen DeLorme.

<div class="image-slide-gallery">

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/Metadata.png"

Reviewer 3: So I understand, how does the user get to this screen? I feel like I saw 3 share buttons only under the QR code but no "settings" button. They could hit back button to get back to home screen, but I'm not sure where they would go from there.

Author: By tapping on the amount / fees copy under the QR code. I will make this more clear. Good catch

   alt-text = "Screen showing the users home page with a number pad that can be used to initiate a request."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/HomeBalance@2x.png"
   layout = "shadow"
   caption = "Avoid showing the users balance alongside the requesting entry point to prevent privacy and security issues."
   alt-text = "Screen showing home screen showing the users balance."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Entering an amount

Once the user is on the requesting entry point, the next thing to do is add an amount to be requested. Users should be able easily toggle between entering amounts in sats, bitcoin, or the users local currency by single tapping the amount field. More on entering amounts on our [Units & Symbols page]({{ "/guide/designing-products/units-and-symbols/" | relative_url }}).

> Users can also skip entering an amount, as there are scenarios where the sender is deciding the amount to send. No amount requests also limit both parties exposure to price volatility as the bitcoin price in fiat terms could change between requesting and the payment being sent.

Reviewer 2: ```suggestion
Users can also skip entering an amount, as there are scenarios where the sender is deciding the amount to send. Requests without an amount also limit both parties exposure to price volatility, as the bitcoin price in fiat terms could change between requesting and the payment being sent.
```

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/HomeEuro@2x.png"
   layout = "shadow"
   caption = "Let users request in fiat currency."
   alt-text = "Screen showing home screen with a fiat euro input selected."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC.png"

Reviewer 2: The "delete character" icon on the left screen is different than the one on the right screen.

Author: Will update, missed that

   retina = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/HomeBTC@2x.png"
   layout = "shadow"
   caption = "Let users request in bitcoin."
   alt-text = "Screen showing home screen with a bitcoin input selected."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Creating the request

> Once an amount to be requested is decided, the next action a user will take is creating the request. When the user clicks the request button the next screen they see is the generated payment request.

Reviewer 2: ```suggestion
Once an amount is decided, the user clicks the `Request` button and is navigated to the next screen for sharing the generated payment request.
```
`Request` appears 4 times in 2 sentences. Trying to reduce.

> This wallet uses a [unified payment request]({{ "/guide/how-it-works/payment-request-formats/#unified-payment-requests" | relative_url }}). These combine a [lightning invoice]({{ "/guide/how-it-works/payment-request-formats/#invoice" | relative_url }}) and an [on-chain address]({{ "/guide/how-it-works/payment-request-formats/#addresses" | relative_url }}) into a single payment request using BIP21. 

Reviewer 2: ```suggestion
This wallet defaults to a [unified payment request]({{ "/guide/how-it-works/payment-request-formats/#unified-payment-requests" | relative_url }}). These combine a [lightning invoice]({{ "/guide/how-it-works/payment-request-formats/#invoice" | relative_url }}) and an [on-chain address]({{ "/guide/how-it-works/payment-request-formats/#addresses" | relative_url }}) into a single payment request using BIP21. 
```

Unified requests remove the friction of users having to choose between requesting with lightning or on-chain, which can be confusing, especially for new users. 

> The on-chain addresses used in this wallet are swap-addresses. Swap-addresses move received bitcoin into the users payment channels in a trust-minimized, non-custodial manner using [submarine swaps](https://thebitcoinmanual.com/articles/btc-submarine-swaps/). This is so users can avoid having both an on-chain and lightning balance. We cover these more in our [lightning services]({{ "/guide/how-it-works/lightning-services/#swaps" | relative_url }}) and receiving page.

Reviewer 2: ```suggestion
The on-chain addresses used in this wallet are swap-addresses. They move received bitcoin into the users payment channels in a trust-minimized, non-custodial manner using [submarine swaps](https://thebitcoinmanual.com/articles/btc-submarine-swaps/). This is so users can avoid having both an on-chain and lightning balance. We cover these more in our [lightning services]({{ "/guide/how-it-works/lightning-services/#swaps" | relative_url }}) and receiving page.
```

> As unified requests [aren't widely supported yet](https://bitcoinqr.dev/), and users may want to request just from lightning or on-chain, they can share the lightning invoice or on-chain address independently. 

Reviewer 2: ```suggestion
As unified requests [aren't widely supported yet](https://bitcoinqr.dev/), and users may want to request just from lightning or on-chain, they have options to share the lightning invoice or on-chain address independently. 
```
`Can` seemed a bit vague.

{% include tip/close.html %}

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Unified.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Unified@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Unified@2x.png"
   layout = "shadow"
   caption = "Unified payment requests combine a lightning and on-chain request into a single payment request."
   alt-text = "Screen showing a unified payment request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Invoice.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   layout = "shadow"
   caption = "Let users share the lightning invoice independently."
   alt-text = "Screen showing an invoice that is part of a unified request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/address.png"

Reviewer 2: Not clear how this screen is so different from above? Shouldn't all these fields be also shown further up the page?

`Name` is probably not needed. That's what you would handle via contact selection.

Author: Name is the requesters name which would use something like NameDesc with BOLT 11 invoices or with BOLT 12 offers there is a name field that acts the same, can remove though

- A fee is charged if the user is requesting an amount higher than their receive limit.
- No fee is charged if the user is requesting an amount lower than their receive limit.
 
<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   layout = "shadow"
   caption = "Let users know if fees will be charged."
   alt-text = "Screen showing a payment request with setup fees shown."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/SetupFee.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/SetupFee@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/SetupFee@2x.png"
   layout = "shadow"

>    caption = "Communicating fees if the user has no channel."

Reviewer 2: Description and contact should also be assignable here. Screen also needs the pagination dots, otherwise it's not consistent with the first screen.

Reviewer 2: Let's also keep currency formatting consistent, euro are used further up the page. `$5.32` => `5.32 €`

Payment addresses are not yet widely supported. If supporting these it should be clear to users that they are experimental and that not all wallets may be able to send to the payment address chosen.

{% include /tip/close.html %}

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddress@2x.png"
   layout = "shadow"
   caption = "Payment addresses should are managed in the settings."
   alt-text = "Screen showing the users settings with a QR icon in the top left that opens a payment address."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddressQR@2x.png"
   layout = "shadow"

>    caption = "This wallet can quickly open and share a payment address by clicking the QR code in the top left of the settings page."

Reviewer 2: Title should say "Your payment address" for consistency (not "code").

   alt-text = "Screen showing QR code of a payment address."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddresses.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddresses@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/PaymentAddresses@2x.png"
   layout = "shadow"

>    caption = "Allow users to manage payment addresses separately  if your app supports multiple."

Reviewer 2: Let's discuss separately if we want to allow for managing multiple payment address, and whether it should be possible to add ones that this wallet cannot generate (lightning address, LNURL-Pay).

Author: I say we just remove this for now and explore it independently.

{% include tip/close.html %}

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Unified.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Unified@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Unified@2x.png"
   layout = "shadow"
   caption = "Unified payment requests combine lightning and on-chain requests, so users don’t have to worry about which to use."
   alt-text = "Screen showing a unified payment request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Invoice.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Invoice@2x.png"
   layout = "shadow"

>    caption = "Let users share the lightning invoice independently."

Reviewer 2: Small (?) next to "Share lightning invoice" might be good for users to quickly pull up help on what this means.

   alt-text = "Screen showing an invoice that is part of a unified request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/address.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/address@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/address@2x.png"
   layout = "shadow"

>    caption = "Let users share the on-chain address independently."

Reviewer 2: Since this on-chain address is a swap address, the screen can indicate the extra fee.

   alt-text = "Screen showing an on-chain address that is part of a unified request in QR format."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/RequestToggle.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/RequestToggle@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/RequestToggle@2x.png"
   layout = "shadow"
   caption = "❌ Avoid users having to choose between requesting with lightning or on-chain."
   alt-text = "Screen showing a toggle between lightning and on-chain request."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

</div>

## Metadata

<div class="center" markdown="1">

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"

Reviewer 2: Should this image show the details screen?

   retina = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   layout = "float-left-desktop -background -shadow"
   caption = "Let users know if fees will be charged."
   alt-text = "Screen showing a payment request with setup fees shown."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

Metadata is additional information that is part of or can be added to a payment request. This can include an amount, note, or name for the sender. Or for the requester, labels, or [assigning a contact]({{ "/guide/daily-spending-wallet/contacts/#adding-a-contact-to-an-outgoing-invoice" | relative_url }}).

Expiration times are something unique to lightning invoices. We cover modifying these [here]({{ "/guide/daily-spending-wallet/requesting/#invoice-expirations" | relative_url }}).

> Ensure users can [backup]({{ "/guide/daily-spending-wallet/backup-and-recovery/landing-page/" | relative_url }}) this metadata to prevent them from losing their transaction history.

Reviewer 3: ```suggestion
Ensure users can [back up]({{ "/guide/daily-spending-wallet/backup-and-recovery/landing-page/" | relative_url }}) this metadata to prevent them from losing their transaction history.
```

Used as a verb in this context

- A fee is charged if the user has no open channels and needs one opened [on-demand]({{ "/guide/how-it-works/lightning-services/#on-demand-liquidity" | relative_url }}) when the payment is received.
- A fee may be charged when requesting a zero-amount invoice if the received amount is higher than their [inbound liquidity]({{ "/guide/how-it-works/liquidity/" | relative_url }}), also known as their receive limit.
- A fee is charged if the user is requesting an amount higher than their receive limit.
- No fee is charged if the user is requesting an amount lower than their receive limit.
 
<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   retina = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   layout = "shadow"
   caption = "Let users know if fees will be charged."
   alt-text = "Screen showing a payment request with setup fees shown."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/SetupFee.png"

Reviewer 2: Labels are called `Tags` in the activity page, would be good to keep that consistent. Same with the user of the tag and note icons, so people can recognize them immediately. Not a big deal, we can make that consistent when we do the screen review after the content updates.

When denominating invoices in fiat, custom expiries should be used to prevent exposure to price volatility.

If requesting a specific amount denominated in fiat, use a shorter expiry and have the invoice refresh each time it expires. For a wallet designed for in-person payments, a 30 - 60 second expiry works well.

If there is no amount defined on the invoice, the invoice is denominated in bitcoin, or the invoice needs to be shared in a message, then use a longer expiry, like 24 hours.

While it is good to allow users to define their own custom expiries, remember that only advanced users will do this. Choose a sensible default that makes sense for your user-base and how the wallet is intended to be used.

For some background on situations the user might encounter with different invoice expiries, see this [blog post](https://d.elor.me/2022/01/lightning-invoice-expiration-ux-considerations/) from designer Stephen DeLorme.

<div class="image-slide-gallery">

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"

Reviewer 2: The title and paragraph are about invoice expirations, but this screen is about fees? Remove it?

Author: That was showing the wrong image, was suppose to be the settings. Bout to push update

   retina = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   modalImage = "/assets/images/guide/daily-spending-wallet/requesting/Fees@2x.png"
   layout = "shadow"
   caption = "Let users know if fees will be charged."
   alt-text = "Screen showing a payment request with setup fees shown."
   width = 250
   height = 541
   modalWidth = 250
   modalHeight = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/requesting/InvoiceExpiration.png"

Reviewer 2: Don't think you need a dedicated screen just for this one setting. Since the details screen already shows the expiration of the current invoice, that option could just be clickable to define a custom expiration.
